internal/dataplane/kong_client.go

```diff
@@ -413,13 +413,18 @@ func (c *KongClient) Update(ctx context.Context) error {
 	// based on the cache contents, we need to ensure it is not modified during the process.
 	var cacheSnapshot store.CacheStores
 	if c.kongConfig.FallbackConfiguration {
-		// TODO: https://github.com/Kong/kubernetes-ingress-controller/issues/6080
-		// Use TakeSnapshotIfChanged to avoid taking a snapshot if the cache hasn't changed.
+		var newSnapshotHash store.SnapshotHash
 		var err error
-		cacheSnapshot, err = c.cache.TakeSnapshot()
+		cacheSnapshot, newSnapshotHash, err = c.cache.TakeSnapshotIfChanged(c.kongConfig.SnapshotHash)
 		if err != nil {
 			return fmt.Errorf("failed to take snapshot of cache: %w", err)
 		}
+		// Empty snapshot hash means that the cache hasn't changed since the last snapshot was taken.
+		if newSnapshotHash == store.SnapshotHashEmpty {
+			c.logger.V(util.DebugLevel).Info("No configuration change; pushing config to gateway is not necessary, skipping")
+			return nil
+		}
```

Comment:
Please consider adding a `KongClient` unit test that will trigger this code path, ensuring that no config push happens in such a case.